src/manifest.rs

```diff
@@ -99,61 +99,102 @@ pub(crate) struct LintTable {
     #[serde(default, deserialize_with = "deserialize_workspace_key")]
     pub(crate) workspace: bool,
     /// individual `lint_name = ...` entries
-    #[serde(flatten, deserialize_with = "deserialize_into_overridemap")]
-    pub(crate) inner: OverrideMap,
+    #[serde(flatten)]
+    pub(crate) inner: BTreeMap<Identifier, OverrideConfig>,
 }
 
-impl From<LintTable> for OverrideMap {
-    fn from(value: LintTable) -> OverrideMap {
-        value.inner
+impl LintTable {
+    /// Converts this into a stack of `OverrideMap`s, where the top of the stack (later
+    /// indices in the `Vec`) has higher priority than the bottom.
+    pub(crate) fn into_stack(self) -> Vec<OverrideMap> {
+        // use a priority -> OverrideMap BTreeMap, which will be sorted by priority
+        let mut map = BTreeMap::<_, OverrideMap>::new();
+        for (id, config) in self.inner {
+            let (priority, overrides) = match config {
+                OverrideConfig::Shorthand(lint_level) => (
+                    0,
+                    QueryOverride {
+                        lint_level: Some(lint_level),
+                        required_update: None,
+                    },
+                ),
+                OverrideConfig::Both {
+                    level,
+                    required_update,
+                    priority,
+                } => (
+                    priority,
+                    QueryOverride {
+                        lint_level: Some(level),
+                        required_update: Some(required_update),
+                    },
+                ),
+                OverrideConfig::LintLevel { level, priority } => (
+                    priority,
+                    QueryOverride {
+                        lint_level: Some(level),
+                        required_update: None,
+                    },
+                ),
+                OverrideConfig::RequiredUpdate {
+                    required_update,
+                    priority,
+                } => (
+                    priority,
+                    QueryOverride {
+                        lint_level: None,
+                        required_update: Some(required_update),
+                    },
+                ),
+            };
+
+            map.entry(priority).or_default().insert(id, overrides);
+        }
+
+        // this will be sorted by key `priority` in ascending order.
+        // to get the stack semantics of later elements take priority,
+        // we need to reverse this.
```

Comment:
consistency nit:
```suggestion
        // This will be sorted by key `priority` in ascending order.
        // To get the stack semantics of later elements take priority,
        // we need to reverse this.
```